packages/devtools_app/lib/src/debugger/debugger_model.dart

```diff
@@ -278,12 +278,28 @@ class FileNode extends TreeNode<FileNode> {
     final root = FileNode('<root>');
 
     for (var script in scripts) {
-      var parts = script.uri.split('/');
+      final uri = Uri.parse(script.uri);
+      final scheme = uri.scheme;
+      var parts = uri.path.split('/');
 
-      // Look for and handle dotted package names.
-      if (parts.isNotEmpty && parts.first.contains('.')) {
+      // handle google3:///foo/bar
+      if (parts.first.isEmpty) {
```

Comment:
I don't follow why this check is necessary. Should we not always filter empty paths?

Reply:
I'll double check whether this is necessary.

Reply:
Yup - I checked - this is necessary for the google3 url case (and will be the only one w/ empty path segments).